plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/index.tsx

```diff
@@ -11,18 +11,29 @@ export type Props = {
   onDatasetAdd: (dataset: CatalogItem) => void;
 };
 
+export type FilterType = "prefecture" | "fileType" | "tag";
+
 const DatasetsPage: React.FC<Props> = ({ rawCatalog, addedDatasetIds, onDatasetAdd }) => {
   const [selectedDataset, setDataset] = useState<CatalogItem>();
   const [selectedTags, selectTags] = useState<Tag[]>([]);
+  const [filter, setFilter] = useState<FilterType>("prefecture");
 
   const handleOpenDetails = useCallback((data?: CatalogItem) => {
     setDataset(data);
   }, []);
 
+  const handleFilter = useCallback((filter: FilterType) => {
+    setFilter(filter);
+  }, []);
+
   const handleTagSelect = useCallback(
     (tag: Tag) =>
-      selectTags(tags => (tags.includes(tag) ? [...tags.filter(t => t !== tag)] : [...tags, tag])),
-    [],
+      selectTags(tags => {
+        const selected = tags.includes(tag) ? [...tags.filter(t => t !== tag)] : [...tags, tag];
```

Comment:
I think you can just do `tags.filter(t => t !== tag)`. The way you have it now you are basically creating an array just to spread it in to another array.